clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java

```diff
@@ -346,4 +347,85 @@ public AlterConfigsResult alterConfigs(Map<ConfigResource, Config> configs) {
      * @return                The AlterConfigsResult
      */
     public abstract AlterConfigsResult alterConfigs(Map<ConfigResource, Config> configs, AlterConfigsOptions options);
+
+    /**
+     * Change the log directory for the specified replicas. This API is currently only useful if it is used
+     * before the replica has been created on the broker. It will support moving replicas that are already created after
+     * KIP-113 is fully implemented.
+     *
+     * Updates are not transactional so they may succeed for some resources while fail for others. The log directory for
+     * a particular replica is updated atomically.
+     *
+     * This operation is supported by brokers with version 0.11.1.0 or higher.
+     *
+     * @param replicaAssignment  The replicas with their log directory absolute path
+     * @return                   The AlterReplicaDirResult
+     */
+    public AlterReplicaDirResult alterReplicaDir(Map<TopicPartitionReplica, String> replicaAssignment) {
```

Comment:
Discussed with @becketqin offline. I will not add `alterReplicaDir()` API in AdminClient in this patch. I will add `alterReplicaDir()` as a public method in KafkaAdminClient so that this method can be used by ReassignPartitionsCommand in this patch.